awscli/clidriver.py

```diff
@@ -506,7 +506,8 @@ def __call__(self, args, parsed_globals):
             event,
             call_parameters=call_parameters,
             parsed_args=parsed_args,
-            parsed_globals=parsed_globals
+            parsed_globals=parsed_globals,
+            session=self._session,
```

Comment:
I would prefer we do this without modifying clidriver.  In general, I think if there's an option to avoid addings more kwargs to an event we should take that option, provided it's not substantially more complex.

I'd prefer we have a class that stores a reference to the session on `session-initialized`, and then have another method that registers the `after-call` handler if needed.

Reply:
Yeah I like that more, I'll try that out.